### PR TITLE
Create ruleset: StyleCaster.com

### PR DESCRIPTION
**Pull request by injust**

### CODE REVIEW

File: src/chrome/content/rules/StyleCaster.com.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+-->
+
+<ruleset name="StyleCaster.com">
+	<target host="stylecaster.com" />
+	<target host="www.stylecaster.com" />
+	<target host="community.stylecaster.com" />
+	<target host="home.stylecaster.com" />
+	<target host="liveintent.stylecaster.com" />
+	<target host="news.stylecaster.com" />
+
```
Comment (YegorIevlev) [inline src/chrome/content/rules/StyleCaster.com.xml]:
```xml
<securecookie host=".+" name=".+" />
```

File: src/chrome/content/rules/StyleCaster.com.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		- view.email.stylecaster.com
+		- home.stylecaster.com, equivalent to stylecaster.com
+		- liveintent.stylecaster.com, equivalent to p.liadm.com
+		- news-cdn.stylecaster.com
+		- news.stylecaster.com, equivalent to stylecaster.com
+		- sa-cdn.stylecaster.com
+		- search.stylecaster.com
+		- shop.stylecaster.com
+		- stylestudio.stylecaster.com
+		- ua-cdn.stylecaster.com
```
Comment (YegorIevlev) [inline src/chrome/content/rules/StyleCaster.com.xml]: Add returned certificates' Common Names.

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 1 above)
Comment (injust) [reply src/chrome/content/rules/StyleCaster.com.xml]: Added, thanks.

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 2 above)
Comment (injust) [reply src/chrome/content/rules/StyleCaster.com.xml]: Added, thanks.

File: src/chrome/content/rules/StyleCaster.com.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	<target host="liveintent.stylecaster.com" />
+	<target host="news.stylecaster.com" />
+
+	<rule from="^http://(community|home|news)\.stylecaster\.com/"
+		to="https://stylecaster.com/" />
+
+	<rule from="^http://liveintent\.stylecaster\.com/"
+		to="https://p.liadm.com/" />
+
+	<securecookie host=".+" name=".+" />
```
[the remark this answers is not in the document]
Comment (injust) [reply src/chrome/content/rules/StyleCaster.com.xml]: Not sure why I didn't do that...thanks for the heads up.

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 2 above)
Comment (Bisaloo) [reply src/chrome/content/rules/StyleCaster.com.xml]: @YegorIevlev, please stop saying that. This kind of info serves no real purpose and adds a maintenance burden.

You are driving several contributors away from something that was fine before. I prefer the style shown in the [contributing guide](https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md#non-working-hosts).

For securecookie, see [this discussion](https://github.com/EFForg/https-everywhere/issues/13762) that was weighing towards **NOT** automatically adding a rule for them.

**Comment by Bisaloo** [comment]
LGTM otherwise. I will merge once you've updated the top comment.

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 2 above)
Comment (YegorIevlev) [reply src/chrome/content/rules/StyleCaster.com.xml]: @Bisaloo We always used to include this information, but I don't know why it is included.

There is another discussion somewhere, leaning towards using `securecookie` unless there is an actual problem preventing its use. I will try to find it. Currently, there is no standard among contributors. Some include it and some do not.

**Comment by injust** [comment]
@Bisaloo I've gotten into the habit of putting the "equivalent to", since I prefer documenting the rules anyway. Plus I believe I was asked by another collaborator to do that.
If you don't mind, I prefer keeping it in.

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 2 above)
Comment (Bisaloo) [reply src/chrome/content/rules/StyleCaster.com.xml]: > We always used to include this information, but I don't know why it is included.

That was true a long time ago. As far as I can tell, for years now, reviewers have always ignored it (they don't care one way or the other) or asked to remove it.

Comment (YegorIevlev) [reply src/chrome/content/rules/StyleCaster.com.xml]: @Bisaloo For an example of merged PR with CNs, see https://github.com/EFForg/https-everywhere/pull/14928/files. No one asked to remove them.

**Comment by Bisaloo** [comment]
@injust, okay to keep "equivalent to" for now. It would be helpful if you could find out who asked you that.

see also https://github.com/EFForg/https-everywhere/pull/15597#discussion_r193652208

File: src/chrome/content/rules/StyleCaster.com.xml (hunk 2 above)
Comment (Bisaloo) [reply src/chrome/content/rules/StyleCaster.com.xml]: It does not disprove what I said:

> As far as I can tell, for years now, reviewers have always ignored it (they don't care one way or the other) or asked to remove it.

No one asked for this to be added. But you can find examples where reviewers asked for it to be removed.

**Comment by injust** [comment]
@Bisaloo I think I started doing that after https://github.com/EFForg/https-everywhere/pull/15251

**Pull request merged**